src/core/src/pass/manager.cpp

```diff
@@ -89,85 +77,117 @@ class stopwatch {
     bool m_active = false;
     std::chrono::nanoseconds m_last_time = std::chrono::high_resolution_clock::duration::zero();
 };
+
+class Profiler {
+public:
+    Profiler(bool visualize, bool profile_pass_enable)
+            : m_visualize(visualize),
+              m_profile_pass_enable(profile_pass_enable) {}
+
+    void start_timer(const std::string& name) {
+        if (m_profile_pass_enable) {
+            stopwatches[name] = stopwatch();
+            stopwatches[name].start();
+        }
+    }
+
+    void stop_timer(const std::string& name, const std::string& msg) {
+        if (m_profile_pass_enable) {
+            auto& stopwatch = stopwatches.at(name);
+            stopwatch.stop();
+            cout << msg << setw(7) << stopwatch.get_milliseconds() << "ms"
+                 << "\n";
+        }
+    }
+
+    void visualize(const shared_ptr<ov::Model>& model, const std::string& name) {
+        if (m_visualize) {
+            // visualizations and serializations will be named after the outermost function
+            const size_t num_digits_in_pass_index = 3;
+            std::string index_str = std::to_string(m_index++);
+            index_str = std::string(num_digits_in_pass_index - index_str.length(), '0') + index_str;
+            auto base_filename = model->get_name() + std::string("_") + index_str + std::string("_") + name;
+
+            auto file_ext = "svg";
+            ov::pass::VisualizeTree vt(base_filename + std::string(".") + file_ext);
+            vt.run_on_model(model);
+        }
+    }
+
+private:
+    size_t m_index = 0;
```

Comment:
let's make it static, numeration would be consecutive and more user-friendly.

Reply:
done